### PR TITLE
Add NSX-T NAT rule resource and data source

### LINKED ISSUE #623: Create new resources for NSX-T NAT management
This task will track progress working on NSX-T edge gateway NAT

### PR DESCRIPTION
**Pull request by Didainius**
Closes #623 
This PR adds NSX-T NAT rule resource and data source. 

Additionally:
* Fixes binary test `vcd.TestAccVcdOrgVdcNsxt.tf` by patching `getComputeStorageProfiles` to put storage profile name from VDC instead of provider VDC (like the patch in govcd - https://github.com/vmware/go-vcloud-director/pull/378/files#diff-b135ff9ceaf8677fa7425e668030203cc93ade5db71ff57ea376b56af2bc0bc4R221)

Note. Acceptance and binary tests passed with `nsxt` tag on 10.0 (without NSX-T), 10.1, 10.2.0, 10.2.2

### CODE REVIEW

File: website/docs/r/nsxt_nat_rule.html.markdown [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    packets received by an organization VDC network that are coming from an external network or from
+    another organization VDC network.
+  * `NO_DNAT` rule prevents the translation of the external IP address of packets received by an
+    organization VDC from an external network or from another organization VDC network.
+  * `SNAT` rule translates the source IP address of packets sent from an organization VDC network
+    out to an external network or to another organization VDC network.
+  * `NO_SNAT` rule prevents the translation of the internal IP address of packets sent from an
+    organization VDC out to an external network or to another organization VDC network.
+
+* `external_addresses` (Optional) IP address or CIDR of external network.
```
Comment (lvirbalas) [inline website/docs/r/nsxt_nat_rule.html.markdown:119]: My main concern of this PR. I have noticed inconsistency in using plural vs. singular notions for addresses:
- The fields `external_addresses`, `internal_addresses` and `snat_destination_addresses` are plural.
- Documentation comments for them are singular use "IP address".
- All the tests and docs are providing only a single IP address.

File: vcd/datasource_vcd_nsxt_nat.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Enables or disables this rule",
+			},
+			"firewall_match": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "VCD 10.2.2+ Determines how the firewall matches the address during NATing if firewall stage is not skipped. Below are valid values.",
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/datasource_vcd_nsxt_nat.go]: Fixed in resource and datasource.

File: vcd/resource_vcd_nsxt_nat_rule.go [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Enables or disables this rule",
+			},
+			"firewall_match": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "VCD 10.2.2+ Determines how the firewall matches the address during NATing if firewall stage is not skipped. Below are valid values.",
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_nsxt_nat_rule.go]: Fixed in resource and datasource.

File: website/docs/r/nsxt_nat_rule.html.markdown [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  traffic to a specific domain, enter an IP address for this domain or an IP address range in CIDR
+  format. If you leave this text box blank, the SNAT rule applies to all destinations outside of the
+  local subnet.
+* `logging` (Optional) - Enable to have the address translation performed by this rule logged
+  (default `false`). **Note** User might lack rights (**Organization Administrator** role by default
+  is missing **Gateway -> Configure System Logging** right) to enable logging, but API does not
+  return error and it is not possible to validate it. `terraform plan` might show difference on
+  every update.
+
+
```
[the remark this answers is not in the document]
Comment (Didainius) [reply website/docs/r/nsxt_nat_rule.html.markdown]: Removed.

File: website/docs/r/nsxt_nat_rule.html.markdown [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+* `firewall_match` (Optional, VCD 10.2.2+) - You can set a firewall match rule to determine how
+  firewall is applied during NAT. One of `MATCH_INTERNAL_ADDRESS`, `MATCH_EXTERNAL_ADDRESS`,
+  `BYPASS`
+
+  * `MATCH_INTERNAL_ADDRESS` - applies firewall rules to the internal address of a NAT rule
+  * `MATCH_EXTERNAL_ADDRESS` - applies firewall rules to the external address of a NAT rule
+  * `BYPASS` - skip applying firewall rules to NAT rule
+
+
```
[the remark this answers is not in the document]
Comment (Didainius) [reply website/docs/r/nsxt_nat_rule.html.markdown]: Removed.

File: website/docs/r/nsxt_nat_rule.html.markdown [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+  * `DNAT` rule translates the IP address and optionally the port (using `dnat_external_port`) of
+    packets received by an organization VDC network that are coming from an external network or from
+    another organization VDC network.
+  * `NO_DNAT` rule prevents the translation of the external IP address of packets received by an
+    organization VDC from an external network or from another organization VDC network.
+  * `SNAT` rule translates the source IP address of packets sent from an organization VDC network
+    out to an external network or to another organization VDC network.
+  * `NO_SNAT` rule prevents the translation of the internal IP address of packets sent from an
+    organization VDC out to an external network or to another organization VDC network.
+
```
[the remark this answers is not in the document]
Comment (Didainius) [reply website/docs/r/nsxt_nat_rule.html.markdown]: Removed.

File: website/docs/r/nsxt_nat_rule.html.markdown (hunk 1 above)
Comment (Didainius) [reply website/docs/r/nsxt_nat_rule.html.markdown]: Yes. After reviewing UI and docs changing all to be singular `external_address`, `internal_address` and `snat_destination_address`

**Review by lvirbalas — APPROVED** [review]
LGTM! Thanks for the tuning!

**Review by dataclouder — APPROVED** [review]

**Pull request merged**